lib/ProxySQL_MCP_Server.cpp

```
@@ -74,33 +75,26 @@ ProxySQL_MCP_Server::ProxySQL_MCP_Server(int p, MCP_Threads_Handler* h)
-		if (handler->query_tool_handler->init() == 0) {
-			proxy_info("Query Tool Handler initialized\n");
-		}
+	// 2. Query Tool Handler (uses Discovery_Schema directly for two-phase discovery)
+	proxy_info("Initializing Query Tool Handler...\n");
+
+	// Hardcode catalog path to datadir/mcp_catalog.db for stability
+	std::string catalog_path = std::string(GloVars.datadir) + "/mcp_catalog.db";
+
+	handler->query_tool_handler = new Query_Tool_Handler(
+		handler->variables.mcp_mysql_hosts ? handler->variables.mcp_mysql_hosts : "",
+		handler->variables.mcp_mysql_ports ? handler->variables.mcp_mysql_ports : "",
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)